deepwell/src/services/user.rs

```diff
@@ -61,6 +62,78 @@ pub struct UpdateUser {
     avatar_path: Maybe<Option<String>>,
 }
 
+#[derive(Serialize, Debug)]
+pub struct UserIdentityOutput {
+    id: i64,
+    username: String,
+    tinyavatar: Option<String>, // TODO
+    karma: u8,
+    role: String,
+}
+
+impl UserIdentityOutput {
+    pub fn from(user: &UserModel) -> Self {
+        Self {
+            id: user.id,
+            username: user.username.clone(),
```

Comment:
Yeah, I cloned it because hypothetically I don't know what you're going to do with these objects, nor what you're going to do with the `UserModel` afterwards.

Reply:
Well here it's being used when you're building a response, so there shouldn't be anything after that I think.

Reply:
Sure, but it does feel a bit wrong to "mangle" the `UserModel` object because I'm _always_ assuming you're building a response with it and discarding the `UserModel` immediately.

I can already think of a few scenarios where I bet this will come up, because `UserIdentity` in particular is used all over the API.

Reply:
I'm not sure what you mean by "mangle", when you are passing it as owned you're destructing it.

Honestly even if it's used elsewhere, `into()` of owned to owned is still preferable imo, because if you need a second copy just do `model.clone().into()`, which makes the cloning explicit.

Reply:
That solution clones the _entire model_ in order to get a `UserIdentity`, rather than just a few `String` instances. I think it's a unnecessary microoptimization to avoid cloning here.

Reply:
It feels like these UserSomethingOutput structs are all *views* into the underlying UserModel. In that case, would a struct with a lifetime be appropriate here?

That said, I agree that this is not a big deal performance-wise.

Reply:
You're right, I would've used borrowed strings everywhere if I could. I feel like there was some kind of ownership reason for why that wasn't possible, but I don't remember.